cmd/ketch/job_deploy.go

```diff
@@ -0,0 +1,107 @@
+package main
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+
+	"github.com/spf13/cobra"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/yaml"
+
+	ketchv1 "github.com/shipa-corp/ketch/internal/api/v1beta1"
+)
+
+const jobDeployHelp = `
+Deploy a job.
+`
+
+const (
+	defaultJobVersion       = "v1"
+	defaultJobParallelism   = 1
+	defaultJobCompletions   = 1
+	defaultJobBackoffLimit  = 6
+	defaultJobRestartPolicy = "Never"
+)
+
+func newJobDeployCmd(cfg config, out io.Writer) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "deploy [FILENAME]",
+		Short: "Deploy a job.",
+		Long:  jobDeployHelp,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			filename := args[0]
+			return jobDeploy(cmd.Context(), cfg, filename, out)
+		},
+	}
+	return cmd
+}
+
+func jobDeploy(ctx context.Context, cfg config, filename string, out io.Writer) error {
+	b, err := os.ReadFile(filename)
+	if err != nil {
+		return err
+	}
+	var spec ketchv1.JobSpec
+	err = yaml.Unmarshal(b, &spec)
+	if err != nil {
+		return err
+	}
+	setJobSpecDefaults(&spec)
+	if err = validateJobSpec(&spec); err != nil {
+		return err
+	}
+
+	var job ketchv1.Job
+	err = cfg.Client().Get(ctx, types.NamespacedName{Name: spec.Name}, &job)
```

Comment:
how about using
```golang
import (
	ctrl "sigs.k8s.io/controller-runtime"
)

ctrl.CreateOrUdate()
```

https://github.com/kubernetes-sigs/controller-runtime/blob/master/alias.go#L115
?

Reply:
Cool. I didn't even know it existed...until now.